### PR TITLE
TIP-706: Add text area normalizer

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**
**ES Filter Checklist:**
- [x] Add attribute/field normalizer
- [x] Review the PQB filter integration tests
- [x] Add index mapping integration tests 
- [x] Write the filter & Spec
- [x] Update the documentation in `search_specification.rst`

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Ok
| Added Behats                      | -
| Added integration tests           | Ok
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php [hunk 1]
```diff
@@ -13,8 +13,7 @@
 class TextAreaFilterIntegration extends AbstractFilterTestCase
 {
     /** @var string Test newlines in TextArea data */
-    private $rabbitNewLineData = 'Why my rabbit
-                            is the best?';
+    private $rabbitNewLineData = 'Why my rabbit\n is the best?';
```
Comment (SamirBoulil) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php]: 👀  Does not work with the previous version :(

Comment (gplanchat) [reply src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php]: Shouldn't it be double quoted ?

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            'indexing' === $format;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getNormalizedData(ProductValueInterface $productValue)
+    {
+        $cleanedData = str_replace('\r', '', $productValue->getData());
+        $cleanedData = str_replace('\n', '', $cleanedData);
```
Comment (anaelChardan) [inline src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php]: Use PHP_EOL constant please :)

**Review by anaelChardan — APPROVED** [review]
Very small comment :)

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php (hunk 2 above)
Comment (gplanchat) [reply src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php]: Why are `\r` and `\n` in single quotes ? is it on purpose ?

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst [hunk 3]
```diff
@@ -207,7 +216,7 @@ CONTAINS
     'filter' => [
         'query_string' => [
             'default_field' => 'description-text.raw',
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst]: values.
everywhere

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst [hunk 4]
```diff
@@ -314,7 +354,15 @@ Data model
 ~~~~~~~~~~
 .. code-block:: php
 
-  name-varchar: "My product name"
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst:317]: need to fix the doc of the DOES_NOT_CONTAIN maybe

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst (hunk 3 above)
Comment (jjanvier) [reply src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst]: what about locale and channel?

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php (hunk 1 above)
Comment (jjanvier) [reply src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php]: so maybe it means it doesn"t work...

Comment (jjanvier) [reply src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php]: we should dig that

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php [hunk 5]
```diff
@@ -78,81 +79,81 @@ protected function setUp()
...
-        $result = $this->execute([['a_text', Operators::STARTS_WITH, 'my dog']]);
+        $result = $this->execute([['a_text_area', Operators::STARTS_WITH, 'my dog']]);
         $this->assert($result, ['best_dog']);
 
-        $result = $this->execute([['a_text', Operators::STARTS_WITH, 'my cat']]);
+        $result = $this->execute([['a_text_area', Operators::STARTS_WITH, 'my cat']]);
         $this->assert($result, ['best_cat']);
 
-        $result = $this->execute([['a_text', Operators::STARTS_WITH, 'why my rabbit is']]);
+        $result = $this->execute([['a_text_area', Operators::STARTS_WITH, 'Why my rabbit is']]);
```
Comment (jjanvier) [inline src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php]: it should pass with the lowercase no?
maybe we should test both cases

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php (hunk 2 above)
Comment (jjanvier) [reply src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php]: `str_replace(["\r\n", "\r", "\n"],"",$string);`

?

File: src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst (hunk 3 above)
Comment (SamirBoulil) [reply src/Pim/Bundle/CatalogBundle/Elasticsearch/Filter/search_specification.rst]: yep.

File: src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php (hunk 5 above)
Comment (SamirBoulil) [reply src/Pim/Bundle/CatalogBundle/tests/integration/PQB/Filter/TextArea/TextAreaFilterIntegration.php]: Mistake I revert this

File: src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php (hunk 2 above)
Comment (SamirBoulil) [reply src/Pim/Component/Catalog/Normalizer/Indexing/Product/TextAreaNormalizer.php]: Yep it works better ;)

**Review by jjanvier — APPROVED** [review]

**Comment by SamirBoulil** [comment]
https://core-ci.akeneo.com/blue/organizations/jenkins/jjanvier%2Fpim-community-dev/detail/PR-95/5/tests

**Pull request merged**